File: python/cucumber/model/petstore.py
import allure
import json
import requests

headers = {
    'Content-Type': 'application/json',
    'accept': 'application/json'
}


def ping():
    requests.get("https://petstore3.swagger.io/v3/pet")


def get(id):
    url = f"https://petstore3.swagger.io/api/v3/pet/{id}"
    allure.attach(
        f"GET {url}",
        name="Query request",
        attachment_type=allure.attachment_type.TEXT
    )
    result = requests.get(url=url, headers=headers)
    allure.attach(
        result.text,
        name="Query response",
        attachment_type=allure.attachment_type.JSON
    )
    return result


def post(payload):
    url = "https://petstore3.swagger.io/api/v3/pet"
    allure.attach(
        f"POST {url}",
        name="Query request",
        attachment_type=allure.attachment_type.TEXT
    )
    allure.attach(
        json.dumps(payload),
        name="Query payload",
        attachment_type=allure.attachment_type.JSON
    )
    result = requests.post(url=url,
                           data=json.dumps(payload),
                           headers=headers)
    allure.attach(
        result.text,
        name="Query response",
        attachment_type=allure.attachment_type.JSON
    )
    return result


def delete(id):
    url = f"https://petstore3.swagger.io/api/v3/pet/{id}"
    allure.attach(
        f"DELETE {url}",
        name="Query request",
        attachment_type=allure.attachment_type.TEXT
    )
    headers['api_key'] = 'special-key'
    result = requests.delete(url=url, headers=headers)
    allure.attach(
        result.text,
        name="Query response",
        attachment_type=allure.attachment_type.JSON
    )
    return result

File: nodejs/cucumber/model/petstore.js
const assert = require('assert');
const axios = require('axios');


headers = {
    'Content-Type': 'application/json',
    'accept': 'application/json'
};

class Petstore {

  get(id, allure) {
    allure.attach("GET https://petstore.swagger.io/v2/pet/" + id, "text/plain");
    return axios.get("https://petstore.swagger.io/v2/pet/" + id);
  }

  post(payload, allure) {
    allure.attach("POST https://petstore.swagger.io/v2/pet/", "text/plain");
    allure.attach(JSON.stringify(payload), "application/json");
    return axios.post("https://petstore.swagger.io/v2/pet/", payload);
  }

  delete(id, allure) {
    allure.attach("DELETE https://petstore.swagger.io/v2/pet/" + id, "text/plain");
    return axios.delete("https://petstore.swagger.io/v2/pet/" + id);
  }

}

module.exports = Petstore;
